src/v/redpanda/admin/services/security.cc

```diff
@@ -0,0 +1,262 @@
+          "No authentication result found");
+    }
+
+    if (
+      auth_result->get_sasl_mechanism()
+      != security::oidc::sasl_authenticator::name) {
+        throw serde::pb::rpc::unauthenticated_exception(
+          "SASL mechanism is not OIDC");
+    }
+
+    proto::admin::resolve_oidc_identity_response resp;
+    resp.set_principal(ss::sstring(auth_result->get_username()));
+
+    security::oidc::authenticator auth{_controller->get_oidc_service().local()};
+    auto res = auth.authenticate(
+      auth_result->get_password().substr(authz_bearer_prefix.length()));
+
+    if (res.has_error() || !res.has_value()) {
+        throw serde::pb::rpc::internal_exception(
+          "Failed to authenticate OIDC token");
+    }
+
+    // Convert ss::lowres_system_clock::time_point to absl::Time
+    resp.set_expire(
+      absl::FromChrono(
+        std::chrono::system_clock::time_point{
+          res.assume_value().expiry.time_since_epoch()}));
+
+    co_return resp;
+}
+
+seastar::future<proto::admin::refresh_oidc_keys_response>
+security_service_impl::refresh_oidc_keys_local() {
+    auto f = co_await ss::coroutine::as_future<>(
+      _controller->get_oidc_service().invoke_on_all(
+        [](security::oidc::service& s) { return s.refresh_keys(); }));
+
+    if (f.failed()) {
+        auto ex = f.get_exception();
+        auto ex_msg = ssx::sformat("{}", ex);
+        vlog(
+          securitylog.warn,
+          "Failed to refresh OIDC keys on some shards: {}",
+          ex_msg);
+        serde::pb::rpc::error_info info;
+        info.reason = "Failed to refresh OIDC keys";
+        info.metadata.emplace(
+          ss::to_sstring(_proxy_client.self_node_id()), ex_msg);
+        throw serde::pb::rpc::internal_exception(std::move(info));
+    }
+
+    vlog(securitylog.debug, "Successfully refreshed OIDC keys on all shards");
+    co_return proto::admin::refresh_oidc_keys_response{};
+}
+
+seastar::future<proto::admin::refresh_oidc_keys_response>
+security_service_impl::refresh_oidc_keys(
+  serde::pb::rpc::context ctx, proto::admin::refresh_oidc_keys_request) {
+    vlog(
+      securitylog.debug,
+      "Refreshing OIDC keys on node {}",
+      _proxy_client.self_node_id());
+
+    if (proxy::is_proxied(ctx)) {
+        co_return co_await refresh_oidc_keys_local();
+    }
+
+    // Not a proxied request, need to run locally and on other nodes
+    std::vector<ss::future<proto::admin::refresh_oidc_keys_response>> futures;
+    futures.push_back(refresh_oidc_keys_local());
+
+    vlog(
+      securitylog.debug,
+      "Proxying to other nodes",
+      _proxy_client.self_node_id());
+    auto clients = _proxy_client.make_clients_for_other_nodes<
+      proto::admin::security_service_client>();
+    for (auto& client_pair : clients) {
+        auto& [node_id, client] = client_pair;
+        futures.push_back(client.refresh_oidc_keys(ctx, {}));
+    }
+
+    auto results = co_await ss::when_all(futures.begin(), futures.end());
```

Comment:
TBH this code is not performance sensitive, I think it would be fine to not do them in parallel and this code would be simpler